ocaml/xapi/repository.ml

```diff
@@ -525,10 +525,10 @@ let get_repository_handler (req : Http.Request.t) s _ =
         true
       with _ -> false
     in
+    let __context =
```

Comment:
This part (`can_be_authorized` and `internal_repo_access_only`) was added for a case when the access of the repo started to require authentication:
1. a coordinator is updated with the new requirement on authentication and try to authenticate its supporters;
2. but the supporters have not been updated so that they would present credential for authentication as they keep the original mind that the repo can be accessed without authentication.

I don't think this will be a valid case anymore. So the `__context` can be get from the `Xapi_http.with_context` at the beginning directly.